### PR TITLE
Add scheduled_send for sending with absolute timeout

### LINKED ISSUE #901: Add scheduled_send for delayed sends with absolute timeout
Currently, our API only allows sending messages with relative timeout via `delayed_send`.

### PR DESCRIPTION
**Pull request by Neverlord**
Close #901.

### CODE REVIEW

File: libcaf_core/caf/send.hpp [hunk 1]
```diff
@@ -108,6 +110,38 @@ void anon_send(const Dest& dest, Ts&&... xs) {
                   std::forward<Ts>(xs)...);
 }
 
+template <message_priority P = message_priority::normal, class Dest = actor,
+          class Rep = int, class Period = std::ratio<1>, class... Ts>
+detail::enable_if_t<!std::is_same<Dest, group>::value>
+delayed_anon_send(const Dest& dest, std::chrono::duration<Rep, Period> rtime,
+                  Ts&&... xs) {
+  static_assert(sizeof...(Ts) > 0, "no message to send");
```
Comment (josephnoir) [inline libcaf_core/caf/send.hpp]: This check are no longer part of the sender mixin. Does one of the other templates implicitly perform it or why did you add it here but remove it in those functions?

Comment (Neverlord) [reply libcaf_core/caf/send.hpp]: Looks like a simple oversight. Good catch!
